inference-engine/src/mkldnn_plugin/nodes/gather_elements.cpp

```diff
@@ -12,70 +15,82 @@ namespace InferenceEngine {
-        if (axis < 0)
-            axis += dataDims.size();
-        if (axis < 0 || axis >= static_cast<int>(dataDims.size()))
-            IE_THROW() << errorPrefix_ << " has invalid axis attribute: " << axis;
-        axis_ = axis;
-
-        auto& outputData = layer->outData[0];
-        strideAxDst_ = outputData->getTensorDesc().getBlockingDesc().getStrides()[axis_];
-        dstAxDim_ = outputData->getTensorDesc().getDims()[axis_];
-        if (axis_ > 0) {
-            strideAx1Diff_ = inputData->getTensorDesc().getBlockingDesc().getStrides()[axis_ - 1] -
-                    outputData->getTensorDesc().getBlockingDesc().getStrides()[axis_ - 1];
-        }
+        return true;
+    }
 
-        LayerConfig config;
-        DataConfig dataConfig, indicesConfig, outConfig;
-        dataConfig.desc = TensorDesc(dataPrecision, dataDims,
-            inputData->getTensorDesc().getLayoutByDims(dataDims));
-        config.inConfs.push_back(dataConfig);
-        indicesConfig.desc = TensorDesc(Precision::I32, indicesDims,
-            indices->getTensorDesc().getLayoutByDims(indicesDims));
-        config.inConfs.push_back(indicesConfig);
+    explicit GatherElementsImpl(const std::shared_ptr<ngraph::Node>& op) : strideAx1Diff_(0) {
+        try {
+            std::string errorMessage;
+            if (!isSupportedOperation(op, errorMessage)) {
+                IE_THROW(NotImplemented) << errorMessage;
+            }
+            errorPrefix_ = std::string("Layer GatherElements with name '") + op->get_friendly_name() + "'";
+
+            if (op->get_input_size() != 2 || op->get_output_size() != 1)
+                IE_THROW() << errorPrefix_ << " has invalid number of input/output edges.";
+
+            const auto& dataDims = op->get_input_shape(dataIndex_);
+            const auto& indicesDims = op->get_input_shape(indicesIndex_);
+            if (dataDims.size() != indicesDims.size())
+                IE_THROW() << errorPrefix_ << " has invalid input shapes. Inputs 'Data' and 'Indices' must have equal ranks.";
+
+            Precision inDataPrecision = details::convertPrecision(op->get_input_element_type(dataIndex_));
+            if (!MKLDNNPlugin::one_of(inDataPrecision.size(),
+                    sizeof(PrecisionTrait<Precision::I32>::value_type),
+                    sizeof(PrecisionTrait<Precision::I16>::value_type),
+                    sizeof(PrecisionTrait<Precision::I8>::value_type))) {
+                IE_THROW() << errorPrefix_ << " has unsupported 'inputData' input precision: " << inDataPrecision;
+            }
 
-        const auto& outDims = outputData->getTensorDesc().getDims();
-        outConfig.desc = TensorDesc(dataPrecision, outDims,
-                outputData->getTensorDesc().getLayoutByDims(outDims));
-        config.outConfs.push_back(outConfig);
+            Precision indicesPrecision = details::convertPrecision(op->get_input_element_type(indicesIndex_));
+            if (indicesPrecision.is_float()) {
```

Comment:
according to spec only i32 and i64 precision supported

Reply:
Fixed